pkg/controller/atlasproject/teams.go

```diff
@@ -243,3 +267,85 @@ func hasTeamRolesChanged(current []string, desired []v1.TeamRole) bool {
 
 	return len(desiredMap) != 0
 }
+
+type assignedTeamInfo struct {
+	ID    string
+	Roles []string
+}
+
+func canAssignedTeamsReconcile(ctx context.Context, atlasClient mongodbatlas.Client, k8sClient client.Client, protected bool, akoProject *v1.AtlasProject) (bool, error) {
+	if !protected {
+		return true, nil
+	}
+
+	latestConfig := &v1.AtlasProjectSpec{}
+	latestConfigString, ok := akoProject.Annotations[customresource.AnnotationLastAppliedConfiguration]
+	if ok {
+		if err := json.Unmarshal([]byte(latestConfigString), latestConfig); err != nil {
+			return false, err
+		}
+	}
+
+	atlasAssignedTeams, _, err := atlasClient.Projects.GetProjectTeamsAssigned(ctx, akoProject.ID())
+	if err != nil {
+		return false, err
+	}
+
+	if atlasAssignedTeams == nil || atlasAssignedTeams.TotalCount == 0 {
+		return true, nil
+	}
+
+	atlasAssignedTeamsInfo := make([]assignedTeamInfo, 0, atlasAssignedTeams.TotalCount)
+	for _, atlasAssignedTeam := range atlasAssignedTeams.Results {
+		if atlasAssignedTeam != nil {
+			atlasAssignedTeamsInfo = append(
+				atlasAssignedTeamsInfo,
+				assignedTeamInfo{
+					ID:    atlasAssignedTeam.TeamID,
+					Roles: atlasAssignedTeam.RoleNames,
+				},
+			)
+		}
+	}
```

Comment:
nitpick
This seems to be a format conversion. Moving to its own function could made this more readable.

```suggestion
	atlasAssignedTeamsInfo := atlasAssignedTeamsToK8s(atlasAssignedTeams.Results)
```